packages/cli/src/Server.ts

```diff
@@ -498,6 +500,18 @@ class App {
 			// Save the workflow in DB
 			const result = await Db.collections.Workflow!.save(newWorkflowData);
 
+			const { tags } = req.body as { tags: string | undefined };
+
+			if (tags) {
+				const tagIds = tags.split(',');
+				await TagHelpers.createRelations(result.id as string, tagIds);
+				const foundTags = await Db.collections.Tag!.find({
+					select: ['id', 'name'],
+					where: { id: In(tagIds) },
+				});
+				result.tags = foundTags.map(({ id, name }) => ({ id: id.toString(), name }));
```

Comment:
Consolidated validators so that the whole request fails in that case.

Reply:
if I understand this correctly, based on current implementation, it would update the workflow first, then update the tags. and throw an error. I think you need to move validation above the updates

Reply:
Tags validation moved to the top in the latest iteration.